lib/methods/ifwapichecktask.cpp

```diff
@@ -269,18 +175,20 @@ static void DoIfwNetIo(
 
 		for (auto& pv : perfdata) {
 			if (!pv.IsString()) {
-				ReportIfwCheckResult(
-					checkable, cmdLine, cr,
+				cr->SetOutput(
 					"Got bad perfdata value " + JsonEncode(perfdata) + " from IfW API on host '"
-						+ psHost + "' port '" + psPort + "', expected an array of strings",
-					start, end
+						+ psHost + "' port '" + psPort + "', expected an array of strings"
 				);
 				return;
 			}
 		}
+
+		cr->SetPerformanceData(PluginUtility::SplitPerfdata(perfdata->Join(" ")));
```

Comment:
Are you sure you can simplify this that much?

* https://github.com/Icinga/icinga2/pull/9062#issuecomment-1614454146

Reply:
Yes.
```bash
[
    "'total::ifw_cpu::load'=7.118014%;;;0;100 'load15m'=6.643090%;;;0;100 'load1m'=7.931634%;;;0;100 'load3m'=6.509641%;;;0;100 'load5m'=5.564334%;;;0;100",
    "'0::ifw_cpu::load'=11.66598%;;;0;100 'load15m'=14.599008%;;;0;100 'load1m'=14.81294%;;;0;100 'load3m'=9.237693%;;;0;100 'load5m'=8.512385%;;;0;100"
]
```

Joining such perfdata results in a single-line string separated by `" "` just like how the individual array elements multiple perfdata are seperated.

Reply:
Yep, seems to work.